src/App/Header/Header.tsx

```diff
@@ -22,9 +22,9 @@ interface Props {
 const Header: React.FC<Props> = ({ withoutSearch }) => {
   const { t } = useTranslation();
   const appContext = useContext(AppContext);
-  const [isInfoDialogOpen, setOpenInfoDialog] = useState();
-  const [showMobileNavBar, setShowMobileNavBar] = useState();
-  const [showLoginModal, setShowLoginModal] = useState(false);
+  const [isInfoDialogOpen, setOpenInfoDialog] = useState<boolean>(false);
```

Comment:
sometimes when TypeScript can’t infer the type, it makes sense to pass a  generic parameter, but in most of the cases TS can cleverly infer the type for useState, especially when we pass a default value... So I believe that passing a generic parameter here is not really necessary + it's very verbose

Reply:
While I agree, in some `useState` calls they were necessary. So in order to not have diverging coding styles I unified the `useState` declarations in places that were affected. Also while they not seem helpful right now, because the code works, TypeScript is now able to error when you try to initialize a state with the wrong type thanks to the generic usage, after all this PR is all about increasing the type safety 😃

Reply:
While Typescript infers values, it does based on how is initialized. So without type anyone could change to `123` and will be hard to notice.

<img width="744" alt="Screen Shot 2020-07-03 at 7 57 25 AM" src="https://user-images.githubusercontent.com/558752/86436603-e2bf6600-bd02-11ea-8788-d3d37a60865f.png">

while the type, perfectly is noticeable and any contributor is aware can only use booleans and not any other value.

<img width="986" alt="Screen Shot 2020-07-03 at 7 57 10 AM" src="https://user-images.githubusercontent.com/558752/86436599-dfc47580-bd02-11ea-812f-c5deda393ca2.png">

Reply:
Anyway, I could not find a eslint rule for this, so, let's use common sense. Let's use types for complex objects while for native let's Typescript do the magic.

Reply:
And I refer to `useState`. For instance things like `Promise<string>` make totally sense.

Reply:
to add there is a subtle difference in the inferred typings for `useState`
used like this: `useState<boolean>();` the type will be `boolean | undefined` while
`useState<boolean>(true);` the type will be just `boolean`

I just like to be more explicit. e.g.
```typescript
const getValue = () => true;
const [useFlag, setUseFlag] = useState(getValue());
```
if `getValue` returns a `string`, then `useState` will not error even though we expect a `boolean`, it will happily infer the `string` type and will instead error further down below, while if `useState` would have made use of the generic to specify the type, it would have error'd.